R/eval_forecasts.R

```diff
@@ -238,16 +279,47 @@ eval_forecasts <- function(data,
 
     if (summarised) {
 
-      res <- res[, .("interval_score" = mean(interval_score),
-                     "calibration" = ifelse(exists("calibration"), mean(calibration), NA),
-                     "bias" = ifelse(exists("bias"), mean(bias), NA),
-                     "sharpness" = ifelse(exists("sharpness"), mean(sharpness), NA)),
-                 by = c(by, "range")]
+      if (!is.null(quantiles)) {
+        # add quantiles for the scores
+        res[, paste0("interval_score_", quantiles) := as.list(quantile(interval_score,
+                                                                       probs = quantiles,
+                                                                       na.rm = TRUE)),
+            by = c(summarise_by, "range")]
+
+        res[, paste0("calibration_", quantiles) := as.list(quantile(calibration,
+                                                                    probs = quantiles,
+                                                                    na.rm = TRUE)),
+            by = c(summarise_by, "range")]
+
+        res[, paste0("bias_", quantiles) := as.list(quantile(bias,
+                                                             probs = quantiles,
+                                                             na.rm = TRUE)),
+            by = c(summarise_by, "range")]
+
+        res[, paste0("sharpness_", quantiles) := as.list(quantile(sharpness,
+                                                                  probs = quantiles,
+                                                                  na.rm = TRUE)),
+            by = c(summarise_by, "range")]
+      }
+
+      # add standard deviation
+      if (sd) {
+        res[, "interval_score_sd" := sd(interval_score, na.rm = TRUE), by = c(summarise_by, "range")]
+        res[, "bias_sd" := sd(bias, na.rm = TRUE), by = c(summarise_by, "range")]
+        res[, "calibration_sd" := sd(calibration, na.rm = TRUE), by = c(summarise_by, "range")]
+        res[, "sharpness_sd" := sd(sharpness, na.rm = TRUE), by = c(summarise_by, "range")]
+      }
+
+      # summarise by taking the mean and omitting unecessary columns
+      res <- res[, lapply(.SD, mean, na.rm = TRUE),
+                 by = c(summarise_by, "range"),
+                 .SDcols = colnames(res) %like% "calibration|bias|sharpness|interval_score"]
     }
     return(res)
   }
 
-  ## scoring for integer or continuous forecasts
+
+  # Score integer or continuous predictions ------------------------------------
```

Comment:
should sharpness be summarised down here as well rather than above?

Reply:
hmm. Not sure I understand what you mean. Can you explain again? If I see correctly then sharpness etc. for integer/continuous forecasts is computed here and only summarised at the end. The part above was all for quantiles.

Reply:
:shrug: order was not clear to me.